model/Image.php

```diff
@@ -165,6 +165,20 @@ public function forTemplate() {
 	}
 
 	/**
+	 * Gets the relative URL accessible through the web. If forced resampling is 
+	 * enabled the URL will point to an optimised file.
+	 *
+	 * @uses Director::baseURL()
+	 * @return string
+	 */
+	public function getURL() {
+		if ($this->ClassName != 'Image_Cached' && Config::inst()->get('Image', 'force_resample')) {
```

Comment:
You could probably use inheritance to solve the specific subclass behaviour here, rather than trying to hack in exemptions based on specific class names.

How about

```
public function getRealURL() { return parent::getURL(); }
public function getURL() {
  if($this->config()->force_resample) {
    return $this->getFormattedImage();
  } else {
    return $this->getRealURL();
  }
}
```

and then in Image_Cached

```
public function getURL() { return $this->getRealURL(); }
```